gitops/prer/create_gitops_prs.go

```diff
@@ -148,6 +152,7 @@ func main() {
 		for _, target := range targets {
 			log.Println("train", train, "target", target)
 			bin := bazel.TargetToExecutable(target)
+			exec.Mustex("", "bazel", "build", target)
```

Comment:
When `:create_gitops_prs` is run once and there are subsequent changes, i.e. gitops result will be expected to be different, this would lead to noops.

```
exec.Mustex("", bin, "--nopush", "--nobazel", "--deployment_root", gitopsdir)
<--->
2021/04/11 10:26:16 exec.go:22: executing: bazel-bin/apps/deployments/docengineservice/docengineservice-ccweb-preview.gitops --nopush --nobazel --deployment_root /var/folders/qy/kldjxbhx6gs4v1rc2n00q3840000gq/T/gitops790537687
```

Adding `exec.Mustex("", "bazel", "build", target)` ensure that the new gitops target is built with the latest source files.

Reply:
This an intentional behavior. Calling gitops executable directly were done to avoid rebuilding the target. The options the binary were build is out of control of PR tool. Calling plain `bazel build` on the target ins some situations might cause full rebuild.

If there are 2 `:create_gitops_prs` calls during the build ( I'm not sure what is the use case for that ), the pipeline script can run the build in between.

Reply:
It is somewhat confusing without the build step, it would mean that we have to 
```
bazel build //... 
bazel run :create_gitops_prs
```

Otherwise if we just run `bazel run :create_gitops_prs` on a clean git clone, the gitops script file would not exist; e.g.
```
2021/04/13 07:58:46 exec.go:37: ERROR: fork/exec bazel-bin/....xxx.gitops: no such file or directory
```

Because `create_gitops_prs` already calls bazel via `bazel cquery`, I don't think it is a bad idea to also call bazel build for each of the gitops target. If we were to set the design rule to say this program should not call bazel directly and all bazel query happens before the program and pass on the query results, we should definitely refrain from calling bazel build within this program.

The reasons for having this build included would be:
- It makes `create_gitops_prs` more user friendly in manual mode.
- It allows dumb retry. Otherwise, retry has to be `bazel build && bazel run :create_gitops_prs`.

I am fine with removing the `exec.Mustex("", "bazel", "build", target)`.

Reply:
I have couple concerns:

1. Running `bazel build <target>` will not work for all use cases. CI pipeline usually constructs quite elaborate options when calling bazel. To support this use we will have to add ability to pass through all additional build options via `:create_gitops_prs` command line. I'm opposing that.
2. Running `bazel build <target>` might cause rebuild because build configuration on the previous step were different.

Please remove `exec.Mustex("", "bazel", "build", target)`.

Reply:
removed.